articles/mobile-automation-testing-with-appium/index.md

```diff
@@ -0,0 +1,99 @@
+- **Little memory consumption:** The architecture of Appium functions as a proxy between the tool-kit for automation and the test machine.
+  
+Appium is also chosen by testers and developers because of the following additional benefits.
+
+- It allows the test scripts to be executed in a parallel way (simultaneously).
+- It consists of a recording tool for recording test cases and a playback tool for playing such records.
+- Various elements can be tested simultaneously (ie. emulators, real devices, and simulators).
+- It supports cloud-based testing using [testdriod](https://en.wikipedia.org/wiki/Testdroid).
+- It consists of a lively community (google group) that improves the ease and speed of troubleshooting.
+  
+### Appium architecture
+The following diagram shows the architecture of Appium:
+![Appium Architecture](/engineering-education/articles/mobile-automation-testing-with-appium/appium-architecture.jpg)
+[Image Source: Software Testing Help](https://www.softwaretestinghelp.com/what-is-appium/)
+
+The Appium architecture consists of three main components: the Appium client, Appium server, and the end device.
+
+#### Appium client
+This consists of the automation code (scripted). This code is written in a specific programming language (Java, Python, Ruby, or Perl). The script contains the configuration details of the application and mobile device. The scripted code and the configuration details are used for running the application’s test cases.
+
+#### Appium server
+This is an HTTP server that receives command requests (HTTP requests) from the client component (in JSON format). This server uses the Node.js language. The commands are then executed on mobile devices. This component should be installed on the computer before scripting the automation code. The Appium server acts as an intermediary between the Appium Client and the end device. It is connected to platforms such as iOS, Android, and Windows.
+
+#### End device
+This refers to an emulator or a mobile device that is connected to the Appium server. The test cases are executed on this device.
+
+### How Appium works?
+Appium works using the three main components of its architecture (client, server, and end device). When this tool is installed on your computer, it sets up an HTTP server that generates a REST API.
+
+This server collects command requests from the Appium client in JSON format. JSON Wire Protocol plays the role of communicating these requests. These command requests are then executed on the end devices (either Android, Windows, or iOS).
+
+The Appium server uses a test automation framework to execute requests on the user interface of end devices. For example, it uses *XCUI test* to execute commands on iOS. It uses *UI Automator* to execute requests on Android platforms (for newer API versions). After the requests have been executed, the Appium server provides HTTP responses to the client.

```

Comment:
A few suggestions:
- Could you brief about `XCUI test` and `UI Automator`?
- Please describe on how the test automation framework works.
- `to execute requests` Please be more specific on what requests are being executed.